core/include/mbox.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    core_mbox Mailboxes
+ * @ingroup     core
+ * @brief       Mailbox implementation
+ *
+ * @{
+ *
+ * @file
+ * @brief       Mailbox API
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef MBOX_H
+#define MBOX_H
+
+#include "list.h"
+#include "cib.h"
+#include "msg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Static initializer for mbox objects */
+#define MBOX_INIT(queue, queue_size) {{0}, {0}, CIB_INIT(queue_size), queue, NULL}
```

Comment:
Also needs a dynamic initializer e.g.:

```C
static inline void mbox_init(mbox_t *mbox, msg_t queue, unsigned int queue_size)
{
    mbox_t m = MBOX_INIT(queue, queue_size);
    *mbox = m;
}
```

so you can (re-)initialize the variable at other points than variable definition